### PR TITLE
Added signin and signup page with authentication

### PR DESCRIPTION
**Pull request by vrinda-mahajan**
Files to be checked:-

- src/pages/auth/profile.jsx
- src/pages/auth/signin.jsx
- src/pages/auth/signup.jsx
- src/components/navbar/navbar.jsx
- src/contexts/auth-context.jsx

Kindly review my PR.
 
[Live site](https://flora-shop-react-git-auth-vrinda-mahajan.vercel.app/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import "./App.css";
-import { HomePage, ProductPage } from "./pages/index";
+import { Cart, HomePage, ProductPage, Wishlist } from "./pages/index";
```
Comment (arjun-computer-geek) [inline src/App.js]: you can remove index from here

File: src/components/card/product-card/product-card.jsx [hunk 2]
```diff
@@ -1,6 +1,58 @@
+import axios from "axios";
+import { useProduct } from "../../../contexts/product-context";
 
 const ProductCard = ({productDetails}) => {
-    const {img,title,inStock,rating,price,productDesc} = productDetails;
+    const {cart,wishlist,dispatch} = useProduct()
+    const encodedToken = localStorage.getItem("token");
+    const {_id,img,title,inStock,rating,price,productDesc} = productDetails;
+
+    const addToCart = async(productDetails) => {
```
Comment (arjun-computer-geek) [inline src/components/card/product-card/product-card.jsx:10]: you can move all function in different file

**Review by arjun-computer-geek — APPROVED** [review]
Code looks good and cleaner

File: src/pages/auth/signup.jsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                    <h3>Sign-up</h3>
+                    <img className="responsive-img auth-img" src={authImg} alt="authentication" />
+                </div>
+
+                <div className="input-auth input-container">
+                    <label className="input-label" htmlFor="username">Firstname</label>
+                    <input onChange={
+                        (e)=>{setUserDetails((prev)=>(
+                            {...prev,firstName:e.target.value}
+                        ))}} className="input-field auth-field" required type="text" id="Firstname" placeholder="Firstname" />
```
Comment (ApurvaSawant11) [inline src/pages/auth/signup.jsx]: You can extract out `setUserDetails`.

File: src/reducers/product-reducer.jsx [hunk 4]
```diff
@@ -35,13 +37,26 @@ export const productReducer = (state,action) => {
...
+                
+        case "CHANGE_CART":
+            return {...state,cart:action.payload}
+
+        case "CHANGE_WISHLIST":
+            return {...state,wishlist:action.payload}
+
+        case "CHANGE_QTY":
+            return {...state,cart:action.payload}
+
```
Comment (ApurvaSawant11) [inline src/reducers/product-reducer.jsx:54]: You can combine `change_cart` and `change_qty` and give a common name. They are the same.

**Review by ApurvaSawant11 — APPROVED** [review]
The code looks good and UI is also clean.

Just try to center the login form and see if you can reduce the height of the signup form so that it fits the viewport height.

File: src/App.js (hunk 1 above)
Comment (vrinda-mahajan) [reply src/App.js]: It is already removed.

File: src/components/card/product-card/product-card.jsx (hunk 2 above)
Comment (vrinda-mahajan) [reply src/components/card/product-card/product-card.jsx:10]: Thanks for suggesting this I will do it later.

File: src/pages/auth/signup.jsx (hunk 3 above)
Comment (vrinda-mahajan) [reply src/pages/auth/signup.jsx]: I have made the changes.

File: src/reducers/product-reducer.jsx (hunk 4 above)
Comment (vrinda-mahajan) [reply src/reducers/product-reducer.jsx:54]: Thanks for pointing this out. I have made the suggested changes.

**Comment by vrinda-mahajan** [comment]
Thanks, @arjun-computer-geek, and @ApurvaSawant11  for reviewing the PR.

**Pull request merged**